src/translate.py

```diff
@@ -2171,6 +2178,8 @@ def elide_casts_for_store(expr: Expression) -> Expression:
         return elide_casts_for_store(uw_expr.expr)
     if isinstance(uw_expr, Literal) and uw_expr.type.is_int():
         # Avoid suffixes for unsigned ints
+        if uw_expr.elide_cast:
+            return Literal(uw_expr.value, type=expr.type, elide_cast=True)
         return Literal(uw_expr.value, type=Type.intish())
```

Comment:
Generally (always?), `late_unwrap(expr).type` is equivalent to `expr.type`. 
```suggestion
        if uw_expr.elide_cast:
            return expr
        return Literal(uw_expr.value, type=Type.intish())
```
Alternatively, this change would make it so you don't need to set `elide_cast=True` in `GlobalInfo.initializer_for_symbol(...)`
```suggestion
        return replace(uw_expr, elide_cast=True)
```

Reply:
With the replace, I do still need the elide in `initializer_for_symbol` otherwise casts still get included when not using `--no-casts`

```
ActorAnimationEntryS D_80AD57E0[0xA] = {
    { (AnimationHeader* )0x600DE68, 1.0f, 0, -1, 2, 0 },
    { (AnimationHeader* )0x600EE98, 1.0f, 0, -1, 2, 0 },
    { (AnimationHeader* )0x600FD34, 1.0f, 0, -1, 0, 0 },
    { (AnimationHeader* )0x60030EC, 1.0f, 0, -1, 2, 0 },
    { (AnimationHeader* )0x6003D78, 1.0f, 0, -1, 2, 0 },
    { (AnimationHeader* )0x600D52C, 1.0f, 0, -1, 0, 0 },
    { (AnimationHeader* )0x6000A44, 1.0f, 0, -1, 0, 0 },
    { (AnimationHeader* )0x6001EF4, 1.0f, 0, -1, 0, 0 },
    { (AnimationHeader* )0x6002224, 1.0f, 0, -1, 0, 0 },
    { (AnimationHeader* )0x6002CB0, 1.0f, 0, -1, 0, 0 },
};```

Reply:
That output is exactly what I'd expect without --no-casts. Int->pointer casts are sketchy and should be pointed out, not least to suppress compiler warnings.

Reply:
This is specifically for data decomp with known structs. Should literals still have a cast when using directly within known data? Code itself isn't affected at all, no casts are lost there, this is only for data.

Reply:
Yes, even for data the cast is sketchy (indicates that either the type is wrong, or it should be passing a symbol reference rather than a literal, e.g. via undefined_syms.txt), and there will be compiler warnings if you don't cast